admin/bin/check-ps-report-for-corrupt-data.js

```diff
@@ -0,0 +1,380 @@
+}
+
+function checkResponseTime (row, prop) {
+  const val = Number(row[prop])
+  if (typeof val !== 'number') {
+    report(prop, row, val)
+  }
+}
+
+function checkTimeOut (row, prop) {
+  const val = row[prop]
+  if (val !== '' && val !== '1' && val !== '0') {
+    report(prop, row, val)
+  }
+}
+
+function checkTimeOutResponse (row, prop) {
+  const val = row[prop]
+  if (val !== '' && val !== '1' && val !== '0') {
+    report(prop, row, val)
+  }
+}
+
+function checkTimeOutSco (row, prop) {
+  const val = row[prop]
+  if (val !== '' && val !== '1' && val !== '0') {
+    report(prop, row, val)
+  }
+}
+
+function checktLoad (row, prop) {
+  regexTest(row, prop, timeRegex)
+}
+
+function checktFirstKey (row, prop) {
+  regexTest(row, prop, timeRegex)
+}
+
+function checktLastKey (row, prop) {
+  regexTest(row, prop, timeRegex)
+}
+
+function checkOverallTime (row, prop) {
+  const val = Number(row[prop])
+  if (typeof val !== 'number') {
+    report(prop, row, val)
+  }
+}
+
+function checkRecallTime (row, prop) {
+  const val = Number(row[prop])
+  if (typeof val !== 'number') {
+    report(prop, row, val)
+  }
+}
+
+function checkReaderStart (row, prop) {
+  regexTest(row, prop, timeRegex)
+}
+
+function checkReaderEnd (row, prop) {
+  regexTest(row, prop, timeRegex)
+}
+
+function checkDob (row, prop) {
+  const regex = /^\d\d\/\d\d\/\d\d\d\d$/
+  if (!row[prop]) {
+    return report(prop, row, row[prop])
+  }
+  regexTest(row, prop, regex)
+}
+
+function checkGender (row, prop) {
+  const val = row[prop]
+  if (val !== 'M' && val !== 'F') {
+    report(prop, row, val)
+  }
+}
+
+function checkPupilId (row, prop) {
+  const val = row[prop]
+  if (!row.hasOwnProperty(prop)) {
+    return reportMissing(row, prop)
+  }
+  if (val.length !== 13) {
+    report(prop, row, val)
+  }
+}
+
+function checkForename (row, prop) {
+  const val = row[prop]
+  if (!row.hasOwnProperty(prop)) {
+    return reportMissing(row, prop)
+  }
+  if (val.length < 1 || val.length > 128) {
+    report(prop, row, val)
+  }
+}
+
+function checkSurname (row, prop) {
+  const val = row[prop]
+  if (!row.hasOwnProperty(prop)) {
+    return reportMissing(row, prop)
+  }
+  if (val.length < 1 || val.length > 128) {
+    report(prop, row, val)
+  }
+}
+
+function checkFormMark (row, prop) {
+  const val = Number(row[prop])
+  if (!row.hasOwnProperty(prop)) {
+    return reportMissing(row, prop)
+  }
+  if (!(RA.isNumber(val)) && RA.inRange(0, 26, val)) {
+    report(prop, row, row[prop])
+  }
+}
+
+function checkQDisplayTime (row, prop) {
+  const val = Number(row[prop])
+  if (!row.hasOwnProperty(prop)) {
+    return reportMissing(row, prop)
+  }
+  if (!RA.inRange(6, 9, val)) {
+    report(prop, row, row[prop])
+  }
+}
+
+function checkPauseLength (row, prop) {
+  const val = Number(row[prop])
+  if (!row.hasOwnProperty(prop)) {
+    return reportMissing(row, prop)
+  }
+  if (val !== 3) {
+    report(prop, row, row[prop])
+  }
+}
+
+function checkDeviceType (row, prop) {
```

Comment:
same reusability comment with checkDeviceType and checkDeviceTypeModel and further down.
They implementation seems quite the same to me.
Am i missing something?

Reply:
i would argue its much clearer to have a separate method that explicitly dictates what is required for the checking of each row.  otherwise you end up with a maintenance headache when you need to change one subtle thing for a particular case.  this is easy to read, and understand.

Reply:
Code refactored and clarified.